rust/src/adapter/otelstdout.rs

```diff
@@ -1,39 +1,52 @@
-use crate::{
-    adapter::otel_formater::{OtelFormatter, ResourceSpan},
-    Event, Metadata,
-};
-use std::{
-    sync::{
-        atomic::{AtomicUsize, Ordering},
-        Arc,
-    },
-    thread, time,
-};
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::sync::Arc;
+use std::{thread, time};
+
+use crate::adapter::otel_formatter::{self, OtelFormatter, ResourceSpan, Span};
+use crate::adapter::{Adapter, Collector};
+use crate::{add_to_linker, Event, Metadata};
+
+use anyhow::Result;
 use tokio::sync::Mutex;
+use wasmtime::Linker;
+
+pub struct OtelAdapterContainer(Arc<Mutex<OtelStdoutAdapter>>);
+
+pub struct OtelTraceCtx(Collector);
 
-use super::{
-    otel_formater::{self, Span},
-    Adapter, Collector,
-};
+impl OtelTraceCtx {
+    pub async fn set_trace_id(&mut self, id: String) {
+        self.0.set_metadata("trace_id".to_string(), id).await;
+    }
+
+    pub async fn shutdown(&self) {
+        self.0.shutdown().await;
+    }
+}
 
-pub type OtelAdapterContainer = Arc<Mutex<OtelStdoutAdapter>>;
+impl OtelAdapterContainer {
+    pub async fn start<T: 'static>(&self, linker: &mut Linker<T>) -> Result<OtelTraceCtx> {
+        let id = next_id();
+        let events = add_to_linker(id, linker)?;
```

Comment:
The reason I didn't do this before is because this makes wasmtime a dependency of the otel stdout adapter. If that's not a concern now this is a much cleaner way of doing things!

Reply:
How?

I still think we need to make these all conform to Cargo features and we add additional cfg'd implementations based on the features used. 

We can go over this together.